services/spar/src/Spar/Error.hs

```diff
@@ -100,6 +100,7 @@ data SparCustomError
   | SparIdPIssuerInUse
   | SparProvisioningMoreThanOneIdP LT
   | SparProvisioningTokenLimitReached
+  | SparInternalError LT
```

Comment:
We have a catch-all that makes this a little redundant.
- https://github.com/wireapp/wire-server/blob/48f7747c84662302676b5362804c33e04e7e03c3/services/spar/src/Spar/Run.hs#L118
- https://github.com/wireapp/wire-server/blob/48f7747c84662302676b5362804c33e04e7e03c3/libs/wai-utilities/src/Network/Wai/Utilities/Server.hs#L182-L198

I think it makes sense to catch it here, but we should add context:

```suggestion
  | -- | FUTUREWORK(fisx): this could also be caught by 'catchErrors' (see "Spar.Run"), maybe we want to remove this case again?  not sure.
    SparInternalError LT
```

Reply:
Sorry, I don't understand this comment. I added this constructor because I need to throw _something_ from the `SAML2` interpreter in case the inspector doesn't work, and none of the existing error messages seemed like a good fit.

Reply:
My point is that if we don't throw anything here, `catchErrors` will handle it in `Spar.Run`.  Do you disagree with that?  (Also I would assume that this is also what would happen to those errors before this change, no?)

Reply:
Added a slightly changed comment